public/assets/ts/admin/editor.ts

```diff
@@ -40,6 +60,103 @@ class Editor {
     container.appendChild(saveButtonContainer);
   }
 
+  // open the rich text modal
+  private openRichTextModal() {
+    const modal = document.getElementById("modal");
+    modal.style.display = "block";
+  }
+
+  // close the rich text modal
+  private closeRichTextModal() {
+    const modal = document.getElementById("modal");
+    modal.style.display = "none";
+  }
+
+  /*
+  controls the rich text editor
+  - enables the button once a markdown field is clicked
+  - disables it otherwise
+  - saves the new markdown into the data once the new markdown has been saved
+  - saves and restores the expansion state of the json editor 
+  */
+  private onEditorEvent(
+    node: { field?: string; value?: string; path?: (string | number)[] },
+    event: Event
+  ) {
+    if (event instanceof PointerEvent) {
+      const openEditorButton = document.getElementById(
+        "open-editor-btn"
+      ) as HTMLButtonElement;
+      if (node && node.field && node.field.includes("Markdown")) {
+        openEditorButton.classList.replace("disabled", "enabled");
+
+        const delta = this.rtEditor.clipboard.convert(node.value);
+        this.rtEditor.setContents(delta);
+
+        const saveRichTextButton = document.getElementById(
+          "save-rich-text-btn"
+        ) as HTMLButtonElement;
+
+        const cancelRichTextButton = document.getElementById(
+          "cancel-rich-text-btn"
+        ) as HTMLButtonElement;
+
+        saveRichTextButton.onclick = () => {
```

Comment:
We should extract this onclick into a function of its own - something like `onSave`